tests/test_inference_problem.py

```diff
@@ -0,0 +1,399 @@
+        model_response = self._fw(prm, sensors, self._ts)
+        error = {}
+        for sensor in sensors:
+            error[sensor] = model_response[sensor] - self._sensor_data[sensor]
+        return error
+
+
+"""
+###############################################################################
+
+                            MAIN CODE
+
+###############################################################################
+"""
+
+if __name__ == "__main__":
+    # Define the sensor
+    s1, s2, s3 = MySensor("S1", 0.2), MySensor("S2", 0.5), MySensor("S3", 42.0)
+
+    fw = MyForwardModel()
+    prm = fw.parameter_list()
+
+    # set the correct values
+    A_correct = 42.0
+    B_correct = 6174.0
+    prm["A"] = A_correct
+    prm["B"] = B_correct
+
+    np.random.seed = 6174
+    noise_sd1 = 0.2
+    noise_sd2 = 0.4
+
+    def generate_data(N_time_steps, noise_sd):
+        time_steps = np.linspace(0, 1, N_time_steps)
+        model_response = fw(prm, [s1, s2, s3], time_steps)
+        sensor_data = {}
+        for sensor, perfect_data in model_response.items():
+            sensor_data[sensor] = perfect_data + np.random.normal(
+                0.0, noise_sd, N_time_steps
+            )
+        return time_steps, sensor_data
+
+    data1 = generate_data(101, noise_sd1)
+    data2 = generate_data(51, noise_sd2)
+
+    me1 = MyModelError(fw, data1)
+    me2 = MyModelError(fw, data2)
+
+    problem = InferenceProblem()
+    key1 = problem.add_model_error(me1, fw.parameter_list())
+    key2 = problem.add_model_error(me2, fw.parameter_list())
+
+    problem.latent.add("A", "A", key1)
+    problem.latent.add("A", "A", key2)
+    problem.latent.add_by_name("B")
+
+    problem.latent.set_prior("A", NormalDistribution(40, 5))
+    problem.latent.set_prior("B", NormalDistribution(6000, 300))
+
+    tmp = problem([A_correct, B_correct])
+
+    problem.noise.add("exp1_noise", s1, key1)
+    problem.noise.add("exp1_noise", s2, key1)
+    problem.noise.add("exp1_noise", s3, key1)
+    problem.noise.set_prior("exp1_noise", Gamma.FromSD(3 * noise_sd1))
+
+    problem.noise.add("exp2_noise", s1, key2)
+    problem.noise.add("exp2_noise", s2, key2)
+    problem.noise.add("exp2_noise", s3, key2)
+    problem.noise.set_prior("exp2_noise", Gamma.FromSD(3 * noise_sd2))
+
+    vb_problem = VariationalProblem(problem)
+    print(vb_problem.prior_MVN())
+
+    print(problem.noise)
+    print(vb_problem.prior_noise())
+
+    # print(vb_problem([A_correct, B_correct]))
+
+    info = variational_bayes(
+        vb_problem, vb_problem.parameter_prior, vb_problem.noise_prior
+    )
+    print(info)
+
+    print(1.0 / info.noise.mean[0] ** 0.5)
+    print(1.0 / info.noise.mean[1] ** 0.5)
+
+    pymc3_problem = PyMC3Problem(problem)
+    print(pymc3_problem([0, 1, 2, 3]))
+
+    def max_likelihood(numbers):
+        return -pymc3_problem(numbers)
+
+    result = scipy.optimize.minimize(max_likelihood, x0=[A_correct, B_correct, noise_sd1, noise_sd2], method="Nelder-Mead")
```

Comment:
the usual maximum-likelihood does not change the hyperparameters, so noise_sd1 would be given. Otherwise I would guess you always get the sd to approach infinity, because that minimizes the model error and thus maximizes the likelihood.